tests/gui/TestGui.cpp

```diff
@@ -1473,7 +1477,96 @@ void TestGui::testDatabaseSettings()
     QTest::keyClick(transformRoundsSpinBox, Qt::Key_Enter);
     QTRY_COMPARE(m_db->kdf()->rounds(), 123456);
 
+    QVERIFY(autosaveDelayCheckBox->isChecked() == false);
+    autosaveDelayCheckBox->toggle();
+    autosaveDelaySpinBox->setValue(autosaveDelayTestValue);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
+    QTRY_COMPARE(m_db->metadata()->autosaveDelayMin(), autosaveDelayTestValue);
+    QApplication::processEvents();
+
     checkSaveDatabase();
+
+    fileDialog()->setNextFileName(m_dbFilePath);
+    triggerAction("actionDatabaseOpen");
+
+    QApplication::processEvents();
+    triggerAction("actionDatabaseSettings");
+    QTRY_COMPARE(autosaveDelayCheckBox->isChecked(), true);
+    QTRY_COMPARE(autosaveDelaySpinBox->value(), autosaveDelayTestValue);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
```

Comment:
Here you open the settings again to make sure the changed recently options are applied. It is clear and makes sense to be tested but I would:

- add the corresponding comment for better navigation between test cases. 
- optional: close the dialog by `Cancel`.
- remove the `processEvents()` call (line 1497), I tested, it works, or let me know why it is needed, may be I missed something.

To be continued...

Reply:
Add the comments, changed to cancel - it saves a bit of time, removed the processEvents.
will push soon